### PR TITLE
updated documentation in data.py, svd.py, ica.py, and pca.py and fixed some typos

### PR DESCRIPTION
**Pull request by d-v-b**
This PR attempts to clarify the documentation in the factorization classes `thunder.factorization.ica`, `thunder.factorization.pca`, and `thunder.factorization.svd` by adding text in the docstrings for these functions that detail the different options for the kwarg `svdMethod`. 

This PR also attempts to clarify the documentation for statistical methods on the `data` class, e.g., `data.mean()`, by indicating in their docstrings that these methods are computed over keys and that output is returned as an ndarray.

I also fixed a few small typos.

### CODE REVIEW

File: python/thunder/factorization/ica.py [hunk 1]
```diff
@@ -22,8 +22,10 @@ class ICA(object):
         Number of independent components to estimate
 
     svdMethod : string, optional, default = "auto"
-        Which SVD method to use. If set to 'auto',
-        will select preferred method based on dimensionality.
+        Which SVD method to use.
+        If set to 'direct', will compute the SVD directly.
```
Comment (freeman-lab) [inline python/thunder/factorization/ica.py]: replace "directly" with "direct gramian matrix estimation and eigenvector decomposition" here and in the other one

File: python/thunder/factorization/ica.py [hunk 2]
```diff
@@ -22,8 +22,10 @@ class ICA(object):
...
+        If set to 'em', will approximate the SVD using iterative expectation-maximization algorithm.
+        If set to 'auto', will select preferred method based on dimensionality of input data.
```
Comment (freeman-lab) [inline python/thunder/factorization/ica.py]: maybe add that auto means "use 'em' if number of columns is greater than 750"

File: python/thunder/rdds/data.py [hunk 3]
```diff
@@ -459,7 +459,7 @@ def count(self):
 
     def mean(self, dtype='float64', casting='safe'):
         """
-        Mean of values, ignoring keys
+        Mean of values across keys, returned as an ndarray.
```
Comment (freeman-lab) [inline python/thunder/rdds/data.py]: These are better but might be even more clear as "Mean of values computed by aggregating across records, returned as an ndarray". A little funny to say mean across keys, almost implies that we're adding up the keys. You could further add that the result will be the same size as a single record. Seem good? Change the others accordingly?

**Comment by freeman-lab** [comment]
Thanks @d-v-b ! Just left a couple minor suggestions, if those look good, once you change them will merge in.

**Comment by d-v-b** [comment]
integrated your suggestions, let me know what you think!

**Comment by freeman-lab** [comment]
Awesome, LGTM. Thanks @d-v-b !

**Pull request merged**